File: apps/exemplo.py
#!/usr/bin/env python3
"""
Aplicativo de Exemplo para o Python App Launcher
Demonstra como o sistema detecta e executa aplicativos Python
"""

import sys
import time
from datetime import datetime

def main():
    """Função principal do aplicativo de exemplo"""
    print("=" * 50)
    print("Python App Launcher - Aplicativo de Exemplo")
    print("=" * 50)
    print(f"Data e hora: {datetime.now().strftime('%d/%m/%Y %H:%M:%S')}")
    print(f"Python versão: {sys.version}")
    print(f"Executado a partir de: {sys.executable}")
    print("=" * 50)
    
    print("\nEste é um aplicativo de exemplo que demonstra:")
    print("✓ Detecção automática de linguagem Python")
    print("✓ Execução segura pelo App Launcher")
    print("✓ Funcionamento correto do sistema")
    
    print("\nAguardando 3 segundos para demonstração...")
    for i in range(3, 0, -1):
        print(f"Fechando em {i}...")
        time.sleep(1)
    
    print("\n✅ Aplicativo executado com sucesso!")
    print("O Python App Launcher está funcionando perfeitamente!")

if __name__ == "__main__":
    main() 

File: apps/JavaScript/exemplo.js
#!/usr/bin/env node
/**
 * Aplicativo JavaScript de Exemplo para o Python App Launcher
 * Demonstra como o sistema detecta e executa aplicativos Node.js
 */

const os = require('os');
const path = require('path');

function main() {
    console.log("=".repeat(50));
    console.log("Python App Launcher - Aplicativo JavaScript de Exemplo");
    console.log("=".repeat(50));
    console.log(`Data e hora: ${new Date().toLocaleString('pt-BR')}`);
    console.log(`Node.js versão: ${process.version}`);
    console.log(`Sistema operacional: ${os.platform()} ${os.release()}`);
    console.log(`Arquitetura: ${os.arch()}`);
    console.log(`Executado a partir de: ${process.execPath}`);
    console.log("=".repeat(50));
    
    console.log("\nEste é um aplicativo JavaScript que demonstra:");
    console.log("✓ Detecção automática de linguagem JavaScript (Node.js)");
    console.log("✓ Execução segura pelo App Launcher");
    console.log("✓ Funcionamento correto do sistema Node.js");
    
    console.log("\nInformações do sistema:");
    console.log(`- CPUs: ${os.cpus().length}`);
    console.log(`- Memória total: ${Math.round(os.totalmem() / 1024 / 1024 / 1024)} GB`);
    console.log(`- Memória livre: ${Math.round(os.freemem() / 1024 / 1024 / 1024)} GB`);
    console.log(`- Diretório atual: ${process.cwd()}`);
    
    console.log("\n✅ Aplicativo JavaScript executado com sucesso!");
    console.log("O Python App Launcher está funcionando perfeitamente com Node.js!");
}

main(); 